checker.cpp

```diff
@@ -1,46 +1,70 @@
-#include <iostream>
-#include <assert.h>
-#include <unistd.h>
+#include "checker.h"
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int VitalsMonitor::vitalsOk(float temperature, float pulseRate, float spo2) {
+    int status = VitalsMonitor::bodyTemperatureOk(temperature);
+    if (0 != status) {
+        status = VitalsMonitor::pulseRateOk(pulseRate);
+        if(0 != status) {
+            status = VitalsMonitor::oxygenSaturationOk(spo2);
+        }
     }
+    return status;
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int VitalsMonitor::bodyTemperatureOk(float temperature) {
+    int status = 1;
+    if(VitalsMonitor::isBodyTemperatureNotOk(temperature)) {
+      cout << "Temperature critical!\n";
+      for (int i = 0; i < 6; i++)
+      {
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
+      }
+      status = 0;
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+    return status;
+}
+
+int VitalsMonitor::pulseRateOk(float pulseRate) {
+    int status = 1;
+    if(VitalsMonitor::isPulseRateNotOk(pulseRate)) {
+      cout << "Pulse Rate is out of range!\n";
+      for (int i = 0; i < 6; i++)
```

Comment:
This for loop is duplicated 3 times. Extract it out into a method, then any changes need to be done only once